analyses/cell-type-ewings/scripts/auc-singler-workflow/04-run-singler.R

```diff
@@ -61,15 +76,32 @@ if (opt$threads > 1) {
   bp_param <- BiocParallel::SerialParam()
 }
 
+# define reference files, if these exist we'll use these rather than downloading
+blueprint_file <- file.path(opt$scratch_dir, "blueprint-celldex.rds")
+hpca_file <- file.path(opt$scratch_dir, "hpca-celldex.rds")
+cl_ont_file <- file.path(opt$scratch_dir, "cl-ont-ref.tsv")
+
+
 # read in files
 sce <- readr::read_rds(opt$sce_file)
 tumor_ref <- readr::read_rds(opt$tumor_reference_file)
 
 # Prep references --------------------------------------------------------------
 
 # grab celldex references
-blueprint_ref <- celldex::BlueprintEncodeData(ensembl = TRUE)
-hpca_ref <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE)
+if (!file.exists(blueprint_file)) {
+  blueprint_ref <- celldex::BlueprintEncodeData(ensembl = TRUE)
+  readr::write_rds(blueprint_ref, blueprint_file)
+} else {
+  blueprint_ref <- readr::read_rds(blueprint_file)
+}
```

Comment:
How much time is this saving? I would expect built-in caching in `celldex` and `ensembldb` to handle most of this most of the time, but maybe I am wrong about that?

Reply:
This wasn't about saving time, it was about being able to run this on LSfR. I was having trouble getting it to download the celldex packages so I saved the file to allow for reuse, but I was also using the old Docker image, so perhaps that was part of the problem and this isn't needed anymore.